itest/send_test.go

```diff
@@ -107,4 +134,65 @@ func testBasicSend(t *harnessTest) {
 		}, defaultTimeout/2)
 		require.NoError(t.t, err)
 	}
+
+	// Close event stream.
+	err = eventNtfns.CloseSend()
+	require.NoError(t.t, err)
+
+	wg.Wait()
+}
+
+// assertRecvSendSateExecEvent asserts that the given send state execution event
+// notification was received. This function will block until the event is
+// received or the event stream is closed.
+func assertRecvSendSateExecEvent(
+	t *harnessTest,
+	eventNtfns tarorpc.Taro_SubscribeSendAssetEventNtfnsClient,
+	targetSendState tarofreighter.SendState,
+) {
+
+	targetSendStateStr := targetSendState.String()
+	foundTargetState := false
+	for {
+		if foundTargetState {
+			break
+		}
+
+		event, err := eventNtfns.Recv()
+
+		// Break if we get an EOF, which means the stream was
+		// closed.
+		//
+		// Use string comparison here because the RPC protocol
+		// does not transport wrapped error structures.
+		if err != nil &&
+			strings.Contains(err.Error(), io.EOF.Error()) {
+
+			break
+		}
+
+		// If err is not EOF, then we expect it to be nil.
+		require.NoError(t.t, err)
+
+		// Check for the transaction broadcast state.
+		switch eventTyped := event.Event.(type) {
+		case *tarorpc.SendAssetEvent_ExecuteSendStateEvent:
```

Comment:
Seems like we may want a default case here that logs the unknown type? Or maybe fails the test.

Reply:
Code at this point will certainly encounter events of types other than `ExecuteSendStateEvent`. I would like it to ignore those types. The goal is to identify a specific type rather than make an assertion on all event types that this code might encounter.